### PR TITLE
Kerberos/SASL support

### PR DESCRIPTION
**Pull request by luke-clifton**
Work in progress: do not merge.

This add support for Kerberos/SASL to `hh`.

### CODE REVIEW

**Comment by jystic** [comment]
:boom: @luke-clifton you're amazing, watching eagerly :)

**Comment by luke-clifton** [comment]
Docker build for centos with gsasl. Place these files in a directory containing hadoop-tools and hadoop-rpc source subdirectories.

https://gist.github.com/luke-clifton/eda2c177b03b313fdc52

**Comment by jystic** [comment]
I might rearrange this so that `hadoop-rpc` is submodule of `hadoop-tools` once I merge your PR to make this kind of thing easier

File: src/Main.hs [hunk 1]
```diff
@@ -61,6 +61,7 @@ main = do
 runHdfs :: forall a. Hdfs a -> IO a
 runHdfs hdfs = handle exitError $ do
     config <- getConfig
+    print config
```
Comment (jberthold) [inline src/Main.hs]: You might want to remove this line before merging J
(I have this running on my Mac via ssh socks proxy, works fine but prints the config every time.)

**Comment by luke-clifton** [comment]
@jystic these tests are obviously not going to pass until hadoop-rpc is updated on Hackage. Do you want to merge this now and then do the submodule thing?

**Comment by jystic** [comment]
Yep, sounds good :+1:

**Pull request merged**